### ISSUE TITLE
[Bug] 7zip files take extremely long to auto extract making extracting larger archives a very long process

### ISSUE THREAD

**Issue by wikewi** [rommapp/playnite-plugin#30]
For some of the bigger, multi file roms/games I use 7z to store them at a higher compression and while they successfully download and do seem to start the unpacking process almost immediately after it starts unpacking it seems to halt and never finish the auto extraction job. I've tried this with a few different archives and on two computers but the results are the same. I'm not really sure if logs are maintained for this add-on if there is I'd be happy to supply more additional information.

**Comment by gantoine** [comment]
There should be an `extensions.log` file generated when errors happen, can you find it and post it here after the issue happens again? https://api.playnite.link/docs/tutorials/extensions/logging.html?tabs=csharp

**Comment by wikewi** [comment]
Thanks for your reply. I have to revise my earlier statement that the extraction never finishes,  because apparently it does but it's  just excruciatingly slow. Also I checked the log but more interestingly there's no mention whatsoever of the extraction process. All I can find is this even after waiting almost 30 minutes for this game to extract. I can tell it did finish because eventually the button changed in playnite from installing to play, it's just that previously I didn't have the patience for it:

14-01 08:47:23.328|DEBUG|RomM#<<Install>b__0>d:Downloading Magical Drop VI to D:\romm\Magical Drop VI\Magical Drop VI.7z.
14-01 08:47:28.276|DEBUG|RomM#<<Install>b__0>d:Download of Magical Drop VI complete.

Note this is not a big game by any means around 700 mb or so zipfile that if I extract it manually it only takes 5 seconds or so on the  same nvme drive which playnite auto extraction also takes place. However for reasons beyond my understanding the auto extract feature in playnite while it works just takes an extermely long time and as the log makes no mention of when the extraction process start or steps I'm afraid there isn't much more I can tell.

**Comment by gantoine** [comment]
Closing this since the issue is slow extraction in playnite and outside the context of the plugin. I'll add some log statements in the next release to update on extraction progress.

**Comment by wikewi** [comment]
Thanks for looking into this. As I'm not very technical I'm assuming the extraction is handled internally by playnite itself and I should mention this issue to the developers of playnite instead?

**Comment by gantoine** [comment]
No the extraction is handled by this plugin, but we defer to the OS to extract the file. The fact that it's slow is likely an issue with the underlying OS, but I have noooooo idea how to fix that with this plugin. Not your OS specifically mind you, just the interop layer between both.

**Comment by wikewi** [comment]
Ok. Just a thought but would it be possible to manually point to 7z.exe executable that can be obtained from 7zip.org and part of it's installation? It seems to be freely available and has the ability to handle most common archive types rar,zip.7z etc

**Comment by gantoine** [comment]
You mean download 7z.exe as part of the install process, or make it a requirement for the romm plugin?

**Comment by wikewi** [comment]
Yeah that would also be a nice option. I've seen a implementation like this on the Extra Metadata Loader plugin (for converting videos but extraction can be a fairly similar processs I guess). Unfortunately the 7zr.exe file on the 7z.org website only handles 7z files but if you install the full version the 7z.exe file installed in your program files\7zip folder should be able to handle all types of archives.

<img width="913" alt="Image" src="https://github.com/user-attachments/assets/642b1951-6fbb-4fd4-9d86-2d1fce6d1ac1" />

**Comment by gantoine** [comment]
I won't be able to do this but I'll re-open the issue if someone else wants to tackle it.

**Comment by adamantike** [comment]
Could this be related to this issue? https://github.com/adamhathcock/sharpcompress/issues/728

**Comment by gantoine** [comment]
This has been released in 0.5.0!
